l3kernel/l3file.dtx

```diff
@@ -3355,6 +3370,24 @@
 % \end{macro}
 % \end{macro}
 %
+% \begin{macro}{\file_input_raw:n, \@@_input_raw:n}
+%   No error checking, no tracking.
+%    \begin{macrocode}
+\cs_new:Npn \file_input_raw:n #1
+  { \exp_args:Ne \@@_input_raw:n { \file_full_name:n {#1} } }
+\cs_new:Npx \@@_input_raw:n #1
+  {
+    \exp_not:N \tl_if_blank:nF {#1}
+      {
+        \exp_not:N \tex_input:D
+          \sys_if_engine_luatex:TF
+            { {#1} }
+            { \exp_not:N \__kernel_file_name_quote:n {#1} \scan_stop: }
+        }
```

Comment:
Can we not (yet) assume a   primitive `\input` takes  braces? (so lose the luatex test?) (needs texlive 2020 or later)

Reply:
I wondered about that: I think we can't yet at like you say it's TL'20, and even moving to TL'19 as a requirement is quite 'up to date'!